radixdlt-core/radixdlt/src/main/java/com/radixdlt/network/messaging/InboundMessage.java

```diff
@@ -65,5 +65,36 @@
 package com.radixdlt.network.messaging;
 
 import com.radixdlt.network.p2p.NodeId;
+import java.util.Arrays;
+import java.util.Objects;
 
-public record InboundMessage(long receiveTime, NodeId source, byte[] message) {}
+public record InboundMessage(long receiveTime, NodeId source, byte[] message) {
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (o == null || getClass() != o.getClass()) return false;
+    InboundMessage that = (InboundMessage) o;
+    return receiveTime == that.receiveTime
+        && source.equals(that.source)
```

Comment:
better to use `Objects.equals` (handles `source` being null)

Reply:
I believed that the auto-generated code in IntelliJ would do it properly.